tests/test_pipeline_mixin.py

```diff
@@ -144,40 +145,84 @@ def run_task_tests(self, task, torch_dtype="float32"):
         model_architectures = self.pipeline_model_mapping[task]
         if not isinstance(model_architectures, tuple):
             model_architectures = (model_architectures,)
-        if not isinstance(model_architectures, tuple):
-            raise TypeError(f"`model_architectures` must be a tuple. Got {type(model_architectures)} instead.")
+
+        # We are going to run tests for multiple model architectures, some of them might be skipped
+        # with this flag we are control if at least one model were tested or all were skipped
+        at_least_one_model_is_tested = False
 
         for model_architecture in model_architectures:
             model_arch_name = model_architecture.__name__
+            model_type = model_architecture.config_class.model_type
 
             # Get the canonical name
             for _prefix in ["Flax", "TF"]:
                 if model_arch_name.startswith(_prefix):
                     model_arch_name = model_arch_name[len(_prefix) :]
                     break
 
-            tokenizer_names = []
-            processor_names = []
+            if model_arch_name not in tiny_model_summary:
+                continue
+
+            tokenizer_names = tiny_model_summary[model_arch_name]["tokenizer_classes"]
+
+            # Sort image processors and feature extractors from tiny-models json file
+            image_processor_names = []
+            feature_extractor_names = []
```

Comment:
if we don't add `None` later in this method (while it's empty), then in `run_model_pipeline_tests`, 

> test_cases = [ ....]

will be empty if `tokenizer_names` is empty or `image_processor_names` is empty or ... (etc).

But we should allow the test case to exist even if only one of them present: for example, a text only model or a vision only model.